features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/aes/aes_alt.c

```diff
@@ -21,298 +21,208 @@
  *  http://csrc.nist.gov/publications/fips/fips197/fips-197.pdf
  */
 
-#if !defined(MBEDTLS_CONFIG_FILE)
-#include "mbedtls/config.h"
-#else
-#include MBEDTLS_CONFIG_FILE
-#endif
+#include "mbedtls/aes.h"
 
 #if defined(MBEDTLS_AES_C)
 #if defined(MBEDTLS_AES_ALT)
 
 #include <string.h>
 
-#include "mbedtls/aes.h"
-
 #include "M480.h"
 #include "mbed_toolchain.h"
 #include "mbed_assert.h"
 
 
 
-#define mbedtls_trace(...) //printf(__VA_ARGS__)
-
 /* Implementation that should never be optimized out by the compiler */
-static void mbedtls_zeroize( void *v, size_t n ) {
-    volatile unsigned char *p = (unsigned char*)v; while( n-- ) *p++ = 0;
+static void mbedtls_zeroize( void *v, size_t n )
+{
+    volatile unsigned char *p = (unsigned char*)v;
+    while( n-- ) *p++ = 0;
 }
 
-
-static uint32_t au32MyAESIV[4] = {
-    0x00000000, 0x00000000, 0x00000000, 0x00000000
-};
-
 extern volatile int  g_AES_done;
```

Comment:
@andresag01 `g_AES_done` is updated in `mbed-os/targets/TARGET_NUVOTON/TARGET_M480/trng_api.c`. Could you suggest a check mechanism? For example, semaphore-like for HAL to avoid busy-wait if it is supported in mbed OS.